WalletWasabi/WabiSabi/WabiSabiCoordinator.cs

```diff
@@ -139,16 +141,46 @@ public void BanDoubleSpenders(object? sender, Transaction tx)
 		// Ban all outputs created by the received transaction because it has spent coins participating in coinjoin rounds.
 		foreach (var indexedOutput in tx.Outputs.AsIndexedOutputs())
 		{
-			Warden.Prison.DoubleSpent(new OutPoint(tx, indexedOutput.N), indexedOutput.TxOut.Value, disruptedRounds);
+			Warden.Prison.DoubleSpent(new OutPoint(tx, indexedOutput.N), indexedOutput.TxOut.Value, disruptedRounds.Select(x => x.RoundId));
 		}
 
-		// Abort disrupted rounds
-		foreach (var roundId in disruptedRounds)
+		// Abort disrupted rounds (only those that pay less than the attacking transaction)
+		using var cts = new CancellationTokenSource(TimeSpan.FromMilliseconds(100));
+		var (succeed, spentCoins) = await GetSpendingCoinsAsync(inputOutPoints, cts.Token).ConfigureAwait(false);
+		var feeRate = succeed ? tx.GetFeeRate(spentCoins) : Constants.AbsurdlyHighFeeRate;
+		var roundsToAbort = disruptedRounds
+			.Where(round => round.MiningFeeRate < feeRate)
+			.Select(x => x.RoundId);
+
+		foreach (var roundId in roundsToAbort)
 		{
 			Arena.AbortRound(roundId);
 		}
 	}
 
+	private async Task<(bool, Coin[])> GetSpendingCoinsAsync(
+		IEnumerable<OutPoint> spendingOutPoints,
+		CancellationToken cancellationToken)
+	{
+		var batch = RpcClient.PrepareBatch();
+		var getTxOutRequests = spendingOutPoints
+			.Select(x => RpcClient.GetTxOutAsync(x.Hash, (int) x.N, true, cancellationToken))
+			.ToList();
+		await batch.SendBatchAsync(cancellationToken).ConfigureAwait(false);
+		var txOutResponses = await Task.WhenAll(getTxOutRequests).ConfigureAwait(false);
```

Comment:
If any of the requests fails we will have an exception here. What happens then? 

https://learn.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.whenall?view=net-8.0

Reply:
Good catch, i forgot it.